.github/workflows/clean_tree.yaml

```diff
@@ -0,0 +1,46 @@
+name: Test Clean Tree
+on:
+  pull_request:
+jobs:
+  git-clean-tree-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Check file existence
+        id: check_files
+        uses: andstor/file-existence-action@v1
+        with:
+          files: "package-lock.json"     
+      - name: Exit When File exists
+        if: steps.check_files.outputs.files_exists == 'true'
+        run: exit 1    
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18     
+      - name: Install yarn
+        run: npm install -g yarn     
+      - name: Install dependencies
+        run: yarn
+      - name: Check Tree after yarn
+        run: |
+          if [[ -n $(git status -s) ]]; then
+            echo "ERROR: tree is dirty after yarn"
+            git status
+            exit 1
+          else
+            echo "Working tree is clean."
+          fi
```

Comment:
do we have to check the tree twice?

Reply:
We don't have to check twice but to make it more clear and understandable what command made the tree to be not clean so the programmer could fix it. However, we can check only once because the yarn command could affect only the yarn.lock file while the yarn build command creates a different files that not related to yarn.lock. What do you prefer?

Reply:
it's up to you, both approaches are valid
thanks for the clarification :+1: :clap:

Reply:
So I leave it as before.

Reply:
You can check for it one time or three time, both options are good.
If you want, you can create a custom github action for this repo that checks the tree is clean and provides a custom error message

Reply:
However, that's an overkill,
Your solution is great and I'm looking forward to see it merge ASAP :pray: 
feel free to merge it now (as is) and upgrade it later
https://github.com/hasadna/open-bus-map-search/pull/180#discussion_r1383601211

Reply:
For now, I am checking it three times. I merged it :)

Reply:
#181 better approach for shell code instead of custom github action